models/meshmodel/registry/registry.go

```diff
@@ -242,7 +242,14 @@ func (rm *RegistryManager) GetModels(db *database.Handler, f types.Filter) ([]v1
 		fmt.Println(modelWithCategoriess)
 		fmt.Println(err.Error()) //for debugging
 	}
-	for _, modelDB := range modelWithCategoriess {
+
+	for num, modelDB := range modelWithCategoriess {
+		filter := &v1alpha1.ComponentFilter{}
+		entities, _, _ := rm.GetEntities(filter)
+		host := rm.GetRegistrant(entities[num])
+		modelDB.HostID = host.ID
+		modelDB.HostName = host.Hostname
```

Comment:
Will you also set `DisplayHostName` here? You can move your logic to convert to Pascal case, (also check, there might be a similar function available already in meshkit)

Reply:
Found it